src/server/AppManager.ts

```diff
@@ -601,6 +603,8 @@ export class AppManager {
             return aff;
         }
 
+        const signature = await this.getAppSignature(descriptor, result.info.id);
+        descriptor.signature = signature;
```

Comment:
Any particular reason to await to a variable to only then assign that variable to a property on an object? Why not the following:
```suggestion
        descriptor.signature = await this.getAppSignature(descriptor, result.info.id);
```